File: frontend/types/department.ts
export type Department = {
  departmentId: number;
  departmentName: string;
  departmentAddress: string | null;
  createdAt: string;
  updatedAt: string | null;
};

export type DepartmentFormValues = {
  departmentName: string;
  departmentAddress: string;
};

File: backend/Models/Department.cs
namespace EmployeeManagement.Api.Models;

public class Department
{
    public int DepartmentId { get; set; }
    public string DepartmentName { get; set; } = string.Empty;
    public string? DepartmentAddress { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public List<Employee> Employees { get; set; } = new();
}
